DocumentVectorPipelineFunctions/BlobTriggerFunction.cs

```diff
@@ -41,20 +42,23 @@ public async Task Run([BlobTrigger("documents/{name}", Connection = "AzureBlobSt
 
     private async Task HandleBlobCreateEventAsync(BlobClient blobClient)
     {
+        var cosmosDBClientWrapper = await CosmosDBClientWrapper.CreateInstance(cosmosClient, this._logger);
+
         this._logger.LogInformation("Analyzing document using DocumentAnalyzerService from blobUri: '{blobUri}' using layout: {layout}", blobClient.Name, "prebuilt-read");
-        var operation = await documentAnalysisClient.AnalyzeDocumentFromUriAsync(
+
+        MemoryStream memoryStream = new MemoryStream();
+        await blobClient.DownloadToAsync(memoryStream);
+
+        var operation = await documentAnalysisClient.AnalyzeDocumentAsync(
             WaitUntil.Completed,
             "prebuilt-read",
-            blobClient.Uri);
+            memoryStream);
 
         var result = operation.Value;
-
         this._logger.LogInformation("Extracted content from '{name}', # pages {pageCount}", blobClient.Name, result.Pages.Count);
 
-        var cosmosDBClientWrapper = await CosmosDBClientWrapper.CreateInstance(cosmosClient, this._logger);
-
         int totalChunksCount = 0;
-        var batchChunkTexts = new List<TextChunk>();
+        var batchChunkTexts = new List<TextChunk>(MaxBatchSize);
```

Comment:
Looks like we still missed clearing `batchChunkTexts` on Line 69, so if we go over `MaxBatchSize` we'll try to upsert the first set of records multiple times.

Reply:
thanks. That's good catch. fixed it.